opennlp-tools/src/main/java/opennlp/tools/util/ext/ExtensionLoader.java

```diff
@@ -41,20 +42,23 @@ static void setOSGiAvailable() {
 
   // Pass in the type (interface) of the class to load
   /**
-   * Instantiates an user provided extension to OpenNLP.
+   * Instantiates a user provided extension to OpenNLP.
    * <p>
    * The extension is either loaded from the class path or if running
    * inside an OSGi environment via an OSGi service.
    * <p>
-   * Initially it tries using the public default
-   * constructor. If it is not found, it will check if the class follows the singleton
-   * pattern: a static field named <code>INSTANCE</code> that returns an object of the type
-   * <code>T</code>.
+   * Initially, the load is conducted using the public no-arg constructor.
+   * If no such constructor is not found, it is checked if the class follows the
+   * {@code Singleton} pattern: a static field named {@code INSTANCE} that
+   * returns an object of the type {@link T}.
    *
-   * @param clazz
-   * @param extensionClassName
+   * @param clazz A reference to {@link Class<T>}.
+   * @param extensionClassName The (fully-qualified) name of the class
+   *                           by which the extension shall be loaded.
    *
    * @return the instance of the extension class
+   *
+   * @throws ExtensionNotLoadedException Thrown if the load operation failed.
    */
   // TODO: Throw custom exception if loading fails ...
```

Comment:
Wonder, if `ExtensionNotLoadedException` isn't the custom exception mentioned in this TODO? The Javadoc states: 

>  Exception indicates that an OpenNLP extension could not be loaded.

Reply:
I was wondering too. Maybe @kinow or @jzonthemtn can clarify here.

Reply:
I don't think that's the exception mentioned, as the TODO was added in this commit where the code was throwing `ExtensionNotLoadedException` - https://github.com/apache/opennlp/commit/21eba24bc2b8ef274251f7c4f146969e7a5a7da0#diff-f8d40ab6a05ba05556cd8b555ec4f60b4c499fdd450e05dc3eb61f9e9d940125R37-R60

Reply:
@kinow Thanks for helping out here.

In discussion with @rzo1, we opted to remove the existing TODO as it is "ancient" in a terms of what the code today does. The catch-block around line 99/100 must be empty, as afterwards another attempt is made to compensate for this via OSGi loading mechanism. Therefore, the `ExtensionNotLoadedException` will be thrown any way, if that compensation attempt also fails. In our (re)view of the code, there is no hard need to introduce other exceptions here. Thus, I'll remove the forgotten TODO.

Reply:
No objections from me. If a user or another dev has a suggestion how to improve it we can always make another TODO or issue. And having a TODO that others don't understand probably confuses more than helps, so +1 to removing it. Thanks